### PR TITLE
[BEAM-14022] Add a logical type for Schema.

### PR DESCRIPTION
**Pull request by laraschmidt**
Adds a logical type for Schemas (so one can use Schemas as a value in a row). Supports conversion to a base type of bytes by converting to a proto and serializing.
R:@TheNeuralBit

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by laraschmidt** [comment]
Run Java PreCommit
Run Java_Examples_Dataflow PreCommit

**Comment by laraschmidt** [comment]
Run Java PreCommit

**Review by TheNeuralBit — APPROVED** [review]

**Comment by laraschmidt** [comment]
Run Java PreCommit

**Comment by TheNeuralBit** [comment]
I will be out next week, but another committer could help merge once the open comment is addressed

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/SchemaTranslation.java [hunk 1]
```diff
@@ -67,10 +68,13 @@
...
   // TODO(BEAM-7855): Populate this with a LogicalTypeRegistrar, which includes a way to construct
-  // the LogicalType
-  // given an argument.
+  // the LogicalType given an argument.
   private static final ImmutableMap<String, Class<? extends LogicalType<?, ?>>>
-      STANDARD_LOGICAL_TYPES = ImmutableMap.of(MicrosInstant.IDENTIFIER, MicrosInstant.class);
+      STANDARD_LOGICAL_TYPES =
+          ImmutableMap.<String, Class<? extends LogicalType<?, ?>>>builder()
+              .put(MicrosInstant.IDENTIFIER, MicrosInstant.class)
+              .put(SchemaLogicalType.IDENTIFIER, SchemaLogicalType.class)
```
[the remark this answers is not in the document]
Comment (laraschmidt) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/SchemaTranslation.java:76]: done. :)

**Pull request merged**